drgn_tools/block.py

```diff
@@ -48,16 +48,30 @@ def for_each_badblocks(bb: Object) -> Iterable[Object]:
         yield (offset, length, ack)
 
 
-def blkdev_ro(bdev: Object) -> bool:
+def blkdev_ro(bdev: Object) -> int:
     """
     Check whether ``struct block_device *`` is read only
 
     :param bdev: ``struct block_device *``
-    :returns: True if readonly
+    :returns: 1 if readonly, 0 if readwrite, -1 if unknown
     """
     if has_member(bdev, "bd_read_only"):
         return bool(bdev.bd_read_only.value_())
-    return bdev.bd_part.policy != 0
+    # "bd_part" is NULL if bdev was not opened yet.
+    if bdev.bd_part:
+        return bdev.bd_part.policy != 0
+    else:
+        return -1
```

Comment:
It looks like blkdev_ro() is used in [drgn_tools/md.py line 140](https://github.com/oracle-samples/drgn-tools/blob/main/drgn_tools/md.py#L140). Could you update that usage? I think it would be as simple as:
```
diff --git a/drgn_tools/md.py b/drgn_tools/md.py
index 6a8f9f9..b94bbf5 100644
--- a/drgn_tools/md.py
+++ b/drgn_tools/md.py
@@ -137,7 +137,7 @@ def show_md_pv(prog: Program, mddev: Object) -> None:
             % (
                 prefix,
                 md_pv_name(rdev),
-                1 if blkdev_ro(rdev.bdev) else 0,
+                blkdev_ro(rdev.bdev),
                 pending,
                 enum_flags_str(prog, "enum flag_bits", rdev.flags),
                 rdev.data_offset.value_(),
```

Reply:
Thanks Stephen. You have have very keen observation. This could have haunt us later. Thanks for your  review. I have fixed this and have also update the typo and have pushed the code for CI testing.